### PR TITLE
Added the sudoku_solver problem in C++

### PR DESCRIPTION
**Pull request by noob-26**
Added the Sudoku solver problem in C++ along with an explanation of the code, proper comments and sample input outputs.

Issue related to this PR #42

### CODE REVIEW

**Comment by raksha009** [comment]
Hey, plz do not create multiple issues for same pr.  I have changed the the title of that issue itself :)

**Comment by noob-26** [comment]
> Hey, plz do not create multiple issues for same pr. I have changed the the title of that issue itself :)

Maam, I created this PR for the Sudoku solver issue and I have also attached this PR to the issue. 
Issue #42

**Comment by raksha009** [comment]
> > Hey, plz do not create multiple issues for same pr. I have changed the the title of that issue itself :)
> 
> Maam, I created this PR for the Sudoku solver issue and I have also attached this PR to the issue.
> Issue #42

Don't worry, we have already acknowledged your Pr with the #42 issue. No need to create multiple issues for referencing the same pr.

File: Code/C++/sudoku_solver.cpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    
+    for(int i = 0; i < 3; ++i)
+    {
+        for(int j = 0; j < 3; ++j)
+        {
+            if(sudoku[submat_row][submat_col] == num)
+                return false;
+        }
+    }
+
```
Comment (raksha009) [inline Code/C++/sudoku_solver.cpp:42]: remove extra line

File: Code/C++/sudoku_solver.cpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            int num;
+            cin>>num;
+            sudoku_row.push_back(num);
+        }
+        sudoku.push_back(sudoku_row);
+    }
+
+    //solving the sudoku by calling the helper method.
+    sudoku_solver(sudoku, 0, 0);
+
```
Comment (raksha009) [inline Code/C++/sudoku_solver.cpp:128]: Remove extra line

File: Code/C++/sudoku_solver.cpp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    return 0;
+}
+
+
+/* 
+Time Complexity of the approach : O(9 ^ (N * N))
```
Comment (raksha009) [inline Code/C++/sudoku_solver.cpp:134]: add time complexity at the end.

**Review by raksha009 — COMMENTED** [review]
Plz link yr file to the resp. folder.

**Comment by noob-26** [comment]
I made the changes to the code and linked the code to the folder.

File: Algorithm/Backtracking/readme.md [hunk 4]
```diff
@@ -28,3 +28,5 @@ unsolvable.
 ## Questions :
 
 * Permutation of a string ----> [C++](/Code/C++/permutation_of_a_string.cpp)
+
```
Comment (raksha009) [inline Algorithm/Backtracking/readme.md:31]: Remove extra line

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Algorithm/Backtracking/readme.md (hunk 4 above)
Comment (noob-26) [reply Algorithm/Backtracking/readme.md]: Made the change

**Review by rudrakshi99 — APPROVED** [review]

File: Code/C++/sudoku_solver.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+4 7 6 8 5 3 9 2 1 
+7 1 9 5 2 8 4 3 6 
+9 6 5 7 3 1 2 8 4 
+2 8 7 4 1 9 3 6 5 
+3 5 4 6 8 2 1 7 9 
+*/
+
+/* 
+Time Complexity of the approach : O(9 ^ (N * N))
+Space Complexity of the approach : O(N * N)
```
Comment (raksha009) [inline Code/C++/sudoku_solver.cpp:187]: Specify n here

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by raksha009** [comment]
Also, plz resolve merge conflicts!

**Comment by noob-26** [comment]
@raksha009 @rudrakshi99 I have made the change and resolved the merge conflict. Please review the changes I made.

**Review by raksha009 — APPROVED** [review]

**Pull request merged**